backend/msgpack/writer.go

```diff
@@ -76,6 +79,8 @@ func newInstanceWriter(instance placement.Instance, opts ServerOptions) instance
 	w := &writer{
 		log:               iOpts.Logger(),
 		metrics:           newWriterMetrics(scope),
+		rand:              rand.New(rand.NewSource(rand.Int63())),
```

Comment:
nit: perhaps just use `rand.NewSource(time.Now().UnixNano())`? not sure if the first number generated by `rand.Int63` will always be the same value.